main.cpp

```diff
@@ -121,385 +163,563 @@ struct CarWash
+{
+    if(sampleIsStereo == true)
+    {
+        outputStereo = true;
+    }
+    else
+    {
+        outputStereo = false;
+    }
+    changePanPosition = 0;
+    panPosition += changePanPosition;
+}
+
+void GlobalSettings::transposeAudio(int changePitch = 2)
+{
+    if(changePitch > 0)
+    {
+        transpose += changePitch;
+    }
+    else 
+    {
+        transpose -= changePitch;
+    }
+}
+
+// UDT 7 // complete
+
+struct Filter 
 {
-    //Frequency cutoff in Hz
     double freqCutoff = 2000.0;
-    //Resonance as a percentage
     float resonance = 80.0f;
-    //Filter type selection
     std::string filterType = "Lowpass";
-    //Distortion amount in dB
-    float distortionAmount = 0.0f;
-    //Key tracker
-    int keyTrackerAmount = 0;
+    float m_distortionAmount = 0.0f;
+    int m_keyTrackerAmount = 0;
 
-    struct CircuitTypeEmulator //second nested UDT
+    struct CircuitTypeEmulator
     {
-        //type selector
         std::string type = "MS20";
-        //drive parameter
         int driveAmount = 0; 
-        //slope of attenuation parameter
         int slope = 24;
-        //self oscillation amount
-        float selfOsc = 20.1f;
-        //toogle switch to clean
+        float selfOscAmount;
         bool isEmulationOn = true;
 
-
-        //create an internal feedback loop
-        void internallyFeedback(float maxFeedbackValue, float selfOsc);
-        //clip audio
-        std::string clipAudio(std::string clipType = "Soft");
-        //manipulate phase
+        void internallyFeedback(float feedbackValue);
+        void clipAudio(std::string clipType);
         void manipulatePhase(bool isLinearPhase);
     };
 
-    //affect frequency content of audio
-    void freqContentAffected(double lowFreq, double highFreq, double freqCutoff);
-    //distort frequency content of audio
+    void freqContentAffected(double lowFreq, double highFreq, double freqCutoffChange);
     void distFreqContent(bool distTurnedUp, float distortionAmount, int driveAmount);
-    //adjust frequency cutoff of filter based on user MIDI note value input
     void midiNoteToCutoff(int midiNoteValue, int keyTrackerAmount);
 
-    CircuitTypeEmulator circuitTypeSelected; //example of member variable whose type is a UDT (CircuitTypeEmulator)
+    CircuitTypeEmulator circuitTypeSelected;
 };
 
-/*
-Thing 8) Envelope
-5 properties:
-    1) Attack time in ms
-    2) Decay time in ms
-    3) Sustain level in dB
-    4) Release time in ms
-    5) Amount of MIDI velocity level as a percentage
-3 things it can do:
-    1) control amplitude level of a sample over time
-    2) control filter cutoff frequency over time 
-    3) control pitch of a sample over time
- */
+void Filter::CircuitTypeEmulator::internallyFeedback(float feedbackValue)
+{
+    if(feedbackValue > 0.7f)
+    {
+        selfOscAmount += feedbackValue;
+    }
+}
+
+void Filter::CircuitTypeEmulator::clipAudio(std::string clipType)
+{
+    clipType = "Soft";
+}
```

Comment:
You are overwriting your function parameter with `"Soft"`.  What is the point of your function parameter if you're not even using it?
You need a member variable that `clipType` can be assigned to.
- [ ] revise